static/js/enketo/widgets/db-object-widget.js

```diff
@@ -77,15 +77,13 @@ define( function( require, exports, module ) {
         var selected = $this.select2('data');
         var doc = selected && selected[0] && selected[0].doc;
         if (doc) {
-            // find the nearest repeat section, or form if not in a repeat
-            var parent = $this.closest('.or-repeat,form.or');
             var field = $this.attr('name');
             var objectRoot = field.substring(0, field.lastIndexOf('/'));
-            updateFields(parent, doc, objectRoot, field);
+            updateFields(doc, objectRoot, field);
         }
     };
 
-    var updateFields = function(parent, doc, objectRoot, keyPath) {
+    var updateFields = function(doc, objectRoot, keyPath) {
```

Comment:
`doc` isn't a great name for this because in recursive calls it's not the document it's just some nested object. I know that's what it was called before but it just confused me...

Reply:
I've changed it to this:
```js
    var updateFields = function(data, keyRoot, originatingKeyPath) {
```
Naming is really hard, and I've really just moved the needle more toward "generic". I toyed with calling `data` `structure` instead, but I didn't think it would work for other readers. Changing `objectRoot` to `keyRoot` also makes more sense to me, because you make your real key with `keyRoot` + '/' + `key`. I also changed `keyPath` ton `originiatingKeyPath` because it's not the current path or anything, it's basically just the root one to not reset (not that I think it even matters: I think resetting it would actually work fine)